log/scope_test.go

```diff
@@ -218,6 +220,130 @@ func TestBasicScopes(t *testing.T) {
 	}
 }
 
+func TestScopeAddLabel(t *testing.T) {
+	const name = "TestScope"
```

Comment:
I am very confused how I am supposed to use the library now. Seems like we have 3 distinct ways to add metadata (scope, label, context) with no clear distinction as to when I should use one or the other

Reply:
See comment about label/context - we can remove one of these if it's going to be too confusing when to use one or the other. 
I think the difference between scope and label/context should be clear though - labels/context are bound to a scope. I agree more examples/docs are needed.

Reply:
removed context for now. we can discuss what to do with it in a chat with doug.